tracer/src/Datadog.Trace/Debugger/LiveDebugger.cs

```diff
@@ -0,0 +1,273 @@
+        {
+            _settings = settings;
+            _discoveryService = discoveryService;
+            _configurationPoller = configurationPoller;
+            _lineProbeResolver = lineProbeResolver;
+            _debuggerSink = debuggerSink;
+            _probeStatusPoller = probeStatusPoller;
+            _unboundProbes = new List<ProbeDefinition>();
+        }
+
+        public static LiveDebugger Instance => LazyInstance.Value;
+
+        private static LiveDebugger Create()
+        {
+            var source = GlobalSettings.CreateDefaultConfigurationSource();
+            var settings = ImmutableDebuggerSettings.Create(DebuggerSettings.FromSource(source));
+            if (!settings.Enabled)
+            {
+                return Create(settings, null, null, null, null, null);
+            }
+
+            var apiFactory = DebuggerTransportStrategy.Get(new ExporterSettings().AgentUri);
+            IDiscoveryService discoveryService = DiscoveryService.Create(source, apiFactory);
+
+            var probeConfigurationApi = ProbeConfigurationApiFactory.Create(settings, apiFactory, discoveryService);
+            var updater = ConfigurationUpdater.Create(settings);
+            IConfigurationPoller configurationPoller = ConfigurationPoller.Create(probeConfigurationApi, updater, settings);
+
+            var snapshotStatusSink = SnapshotSink.Create(settings);
+            var probeStatusSink = ProbeStatusSink.Create(settings);
+
+            var batchApi = BatchUploadApiFactory.Create(settings, apiFactory, discoveryService);
+            var batchUploader = BatchUploader.Create(batchApi);
+            var debuggerSink = DebuggerSink.Create(snapshotStatusSink, probeStatusSink, settings, batchUploader);
+
+            var lineProbeResolver = LineProbeResolver.Create();
+            var probeStatusPoller = ProbeStatusPoller.Create(settings, probeStatusSink);
+
+            return Create(settings, discoveryService, configurationPoller, lineProbeResolver, debuggerSink, probeStatusPoller);
+        }
+
+        // for tests
+        internal static LiveDebugger Create(ImmutableDebuggerSettings settings, IDiscoveryService discoveryService, IConfigurationPoller configurationPoller, ILineProbeResolver lineProbeResolver, IDebuggerSink debuggerSink, IProbeStatusPoller probeStatusPoller)
+        {
+            var debugger = new LiveDebugger(settings, discoveryService, configurationPoller, lineProbeResolver, debuggerSink, probeStatusPoller);
+            debugger.Initialize();
+
+            return debugger;
+        }
+
+        private void Initialize()
+        {
+            if (!_settings.Enabled)
+            {
+                Log.Information("Live Debugger is disabled. To enable it, please set DD_DEBUGGER_ENABLED environment variable to 'true'.");
+                return;
+            }
+
+            Log.Information("Live Debugger initialization started");
+
+            Task.Run(async () => await InitializeAsync().ConfigureAwait(false));
```

Comment:
Before @kevingosse gets to it... 😉 

```suggestion
            Task.Run(() => InitializeAsync());
```